consensus/beacon/consensus.go

```diff
@@ -333,6 +333,39 @@ func (beacon *Beacon) Finalize(chain consensus.ChainHeaderReader, header *types.
 	// The block reward is no longer handled here. It's done by the
 	// external consensus engine.
 	header.Root = state.IntermediateRoot(true)
+	// Open the pre-tree to prove the pre-state against
```

Comment:
This would prevent pre-Cancun blocks from being Finalized (Cancun is the currently planned fork name for verkle trees).

```suggestion
	// Open the pre-tree to prove the pre-state against
	if chain.Config().IsCancun(header.Number) {
```